torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2798,6 +2798,8 @@ Example::
           "split_from", &::c10d::ProcessGroupNCCL::Options::split_from)
       .def_readwrite(
           "split_color", &::c10d::ProcessGroupNCCL::Options::split_color)
+      .def_readwrite(
```

Comment:
Instead of adding an attribute under `c10d::ProcessGroupNCCL::Options`,
you could directly modify the ncclConfig object, like we do for `max_ctas` etc...
https://github.com/pytorch/pytorch/blob/c1e7e40f246ece1c402c11e7ac049dcdaaf6b93a/torch/csrc/distributed/c10d/init.cpp#L2746

```
.def_readwrite("split_share", &ncclConfig_t::splitShare)
```

Reply:
To change default of splitShare to 1, you would need to set it here:
https://github.com/pytorch/pytorch/blob/c1e7e40f246ece1c402c11e7ac049dcdaaf6b93a/torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp#L431
```
config.splitShare = 1;
```
It would also need to be documented that is `1` by default with PyTorch since NCCL sets it to `0` by default.